dev-util/gnome-builder/gnome-builder-46.3.ebuild

```diff
@@ -0,0 +1,274 @@
+
+DESCRIPTION="An IDE for writing GNOME-based software"
+HOMEPAGE="https://wiki.gnome.org/Apps/Builder https://gitlab.gnome.org/GNOME/gnome-builder"
+
+# FIXME: Review licenses at some point
+LICENSE="GPL-3+ GPL-2+ LGPL-3+ LGPL-2+ MIT CC-BY-SA-3.0 CC0-1.0"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="clang doc +d-spy flatpak +git gtk-doc spell +sysprof test +webkit"
+REQUIRED_USE="
+	${PYTHON_REQUIRED_USE}
+	flatpak? ( git )
+"
+
+# When bumping, pay attention to all the included plugins/*/meson.build (and other) build files and the requirements within.
+# `grep -rI dependency * --include='meson.build'` can give a good initial idea for external deps and their double checking.
+# The listed RDEPEND order should roughly match that output as well, with toplevel one first then sorted by file path.
+# Most plugins have no extra requirements and default to enabled; we need to handle the ones with extra requirements. Many of
+# them have optional runtime dependencies, for which we try to at least notify the user via DOC_CONTENTS (but not all small
+# things); `grep -rI -e 'command-pattern.*=' -e 'push_arg'` can give a (spammy) idea, plus python imports in try/except.
+
+# Editorconfig needs old pcre, with vte migrating away, might want it optional or ported to pcre2?
+# An introspection USE flag of a dep is required if any introspection based language plugin wants to use it (grep for gi.repository). Last full check at 3.28.4
+
+# TODO: Handle llvm slots via llvm.eclass; see plugins/clang/meson.build
+RDEPEND="
+	>=dev-libs/glib-2.75:2
+	>=gui-libs/gtk-4.10:4[introspection]
+	>=gui-libs/libadwaita-1.5_alpha:1
+	>=gui-libs/libpanel-1.5.0:1
+	>=gui-libs/gtksourceview-5.8:5[introspection]
+	>=dev-libs/json-glib-1.2.0
+	>=dev-libs/jsonrpc-glib-3.43.0:=
+	>=dev-libs/libdex-0.2:=
+	>=dev-libs/libpeas-1.99.0[python,${PYTHON_SINGLE_USEDEP}]
+	dev-libs/libportal:=[gtk,introspection]
+	>=dev-libs/template-glib-3.36.1:=[introspection]
+	>=gui-libs/vte-0.76.0:2.91-gtk4[introspection]
+	>=dev-libs/libxml2-2.9.0
+	webkit? ( >=net-libs/webkit-gtk-2.38.0:6=[introspection] )
+	>=app-text/cmark-0.29.0:0=
+	d-spy? ( >=dev-debug/d-spy-1.4.0:1 )
+	app-text/editorconfig-core-c
+	flatpak? (
+		dev-util/ostree
+		>=net-libs/libsoup-3:3.0
+		>=sys-apps/flatpak-1.10.2
+	)
+	git? (
+		dev-libs/libgit2:=[ssh,threads]
+		>=dev-libs/libgit2-glib-1.1.0[ssh]
+	)
+
+	>=dev-libs/gobject-introspection-1.74.0:=
+	${PYTHON_DEPS}
+	clang? ( sys-devel/clang:= )
+	spell? (
+		app-text/enchant:2
+		dev-libs/icu:=
+	)
+	sysprof? (
+		>=dev-util/sysprof-45.0[gtk]
+	)
+"
+DEPEND="${RDEPEND}"
+# TODO: runtime ctags path finding..
+
+# desktop-file-utils required for tests, but we have it in deptree for xdg update-desktop-database anyway, so be explicit and unconditional
+# appstream-glib needed for validation with appstream-util with FEATURES=test
+BDEPEND="
+	doc? (
+		$(python_gen_cond_dep '
+			dev-python/sphinx[${PYTHON_USEDEP}]
+			dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]
+		')
+	)
+	gtk-doc? (
```

Comment:
shouldn't this be moved to "doc" USE flag? It seems it migrated away from gtk-doc

Reply:
upstream has both doc and help 

meson_options.txt
```
option('help', type: 'boolean', value: false)
option('docs', type: 'boolean', value: false)
```

gnome-builder-46.3.ebuild
```
        $(meson_use doc help)
        $(meson_use gtk-doc docs)
```

What is the recommended approach for managing USE flags in this case?

Reply:
You are right, probably as it is now (doc -> help, gtk-doc) is ok